src/seq.rs

```diff
@@ -19,6 +19,60 @@ use super::Rng;
 
 #[cfg(not(feature="std"))] use alloc::Vec;
 
+/// Trait to provide convenience function directly on slices rather than
+/// through [`Rng`]. Allows for example calling `vec.choose(&mut rng)` rather
+/// than `rng.choose(&vec)`.
+/// [`Rng`]: trait.Rng.html
+pub trait SliceRandom {
+    #[doc(hidden)]
+    type Item;
+
+    /// Same as [`Rng.choose`].
+    /// [`Rng.choose`]: trait.Rng.html#method.choose
+    fn choose<R>(&self, rng: &mut R) -> Option<&Self::Item>
+        where R: Rng + ?Sized;
+
+    /// Same as [`Rng.choose_mut`].
+    /// [`Rng.choose_mut`]: trait.Rng.html#method.choose_mut
+    fn choose_mut<R>(&mut self, rng: &mut R) -> Option<&mut Self::Item>
+        where R: Rng + ?Sized;
+}
+
+impl<T> SliceRandom for [T] {
+    type Item = T;
+
+    fn choose<'a, R>(&'a self, rng: &mut R) -> Option<&'a Self::Item>
+        where R: Rng + ?Sized
+    {
+        rng.choose(self)
+    }
+
+    fn choose_mut<'a, R>(&'a mut self, rng: &mut R) -> Option<&'a mut Self::Item>
+        where R: Rng + ?Sized
+    {
+        rng.choose_mut(self)
+    }
+}
+
+/// XXX Any way to avoid '+ Sized' here?
```

Comment:
Not in an API that consumes self (i.e. `fn choose(self, ...)`). It's possible to use `where Self: Sized` on each method instead, but in practice I don't think that gains anything (unless we want to add methods taking `&self`, e.g. like `size_hint` does).